### ISSUE TITLE
UndocumentedPublicClass false positive for inner types

### ISSUE THREAD

**Issue by jcornaz** [arturbosch/detekt#2580]
## Expected Behavior

Given the following source code:

```kotlin
private class MyClass {
  object MyInnerObject
}
```

I don't expect the `UndocumentedPublicClass` to be reported because `MyClass` is private.

Yes `MyInnerObject` is not *explicitly* marked private. But being an inner type of a private one, makes `MyInnerObject` *effectively* private.

## Observed Behavior
Given the snippet above and the rule `UndocumentedPublicClass` activated: 

```yaml
  UndocumentedPublicClass:
    active: true
    searchInInnerObject: true
```

A `UndocumentedPublicClass` issue is reported, saying that `MyInnerObject` is public and undocumented.

## Steps to Reproduce
The snippets above should be enough to reproduce the issue.

## Context
I added a private class with inner types and got the mentioned false-positive reported by detekt.

## Your Environment
* Version of detekt used: `1.7.4`
* Version of Gradle used: `6.3`
* Operating System and version: `Linux 5.5.13`

**Comment by BraisGabin** [comment]
I think that we don't even need `searchInInnerObject` that object should never be flagged.

**Comment by schalkms** [comment]
Thanks for reporting this.
The rule doesn't take nested classes into account at the moment.
Following code should be flagged, however, since the nested and outer class(es) are public.

```kotlin
class Outer(val i: Int) {
    // must be documented
    class Nested(val s: String) {
    }
}
```

**Comment by jcornaz** [comment]
@schalkms 

> The rule doesn't take nested classes into account at the moment.

Mmmh. Yes they do... There is even explicit config for it:

https://arturbosch.github.io/detekt/comments.html#configuration-options-2

> Following code should be flagged, however, since the nested and outer class(es) are public.

Yes, it should and is flagged. As far as I can see, that works as expected.

The problem is only when the outer class is not public.
